### PR TITLE
Added "system" agent state type

### PR DESCRIPTION
**Pull request by ajspetner**
*Issue #908 *

*Description of changes:*
Added "system" to the AgentStateType enum.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ajspetner** [comment]
Also added the "error" state type.

File: src/index.d.ts [hunk 1]
```diff
@@ -665,6 +665,12 @@ declare namespace connect {
 
     /** The agent is offline. */
     OFFLINE = "offline",
+
+    /** The agent is in a system-defined state */
+    SYSTEM = "system",
+
+    /** The agent is in an error state */
+    SYSTEM = "error",
```
Comment (vpogudina) [inline src/index.d.ts:673]: Did you mean to have ERROR = "error"

Comment (ajspetner) [reply src/index.d.ts]: Yes, whoops. Just fixed it.

**Comment by vpogudina** [comment]
Fixed it as a part of another pull request, thank you for your contribution.

**Pull request closed without merging**